zio/zjsonio/reader.go

```diff
@@ -239,7 +239,7 @@ func decodeContainer(builder *zcode.Builder, typ zng.Type, body []interface{}) e
 		// each column either a string value or an array of string values
 		if column == nil {
 			// this is an unset column
-			if zng.IsContainerType(childType) || zng.IsContainerType(columns[k].Type) {
+			if zng.IsContainerType(zng.AliasedType(childType)) || zng.IsContainerType(zng.AliasedType(columns[k].Type)) {
```

Comment:
being that we don't seem to care about the alias types here I would do this:
- line 273: `childType = zng.AliasedType(childType)`
- line 253: `childType = zng.AliasedType(childType)`
- drop changes on line 242 and 255 and all references to `realType`.
The variable name `realType` had me confused with whether it is referring to the container Type or the child Type.

Reply:
> being that we don't seem to care about the alias types here I would do this:

Why don't we need it here? I thought we needed to undo aliases wherever we handled types (as we do elsewhere, for example in tzngio). The whole alias scheme remains a bit fuzzy to me though, so I'd be happy to learn here.

> The variable name realType

Ok, I borrowed this name from tzngio, but agree it's not highly descriptive. I'll change the name before merging.

Reply:
> > being that we don't seem to care about the alias types here I would do this:

> Why don't we need it here? I thought we needed to undo aliases wherever we handled types (as we do elsewhere, for example in tzngio). The whole alias scheme remains a bit fuzzy to me though, so I'd be happy to learn here.


@mattnibs I added a test in 0f41a8b2ce22c82c81fed686327dd36464c243c5 that answers this question - without the changes suggested in https://github.com/brimsec/zq/pull/793#discussion_r426793717, this test fails.